Update array_properties.py

biermanncarl opened the pull request:
In the ArrayPropertyTest class, the lbf actor setup was moved to the test_lb() method, which was the only test using the actor. The actor is now removed after the test. This fixes an issue in test_partial_periodic() when changing the system's periodicity. The error which occured can be seen in event.cpp, function on_cell_structure_change(). This, however, didn't cause the test to fail because the error was not handled by Python. It did cause the test to fail during development of the Stokesian Dynamics functionality, when a handle_errors() was added into globals.pyx.

This replaces my two previos change suggestions on the file. Sorry for the confusion.

biermanncarl committed:
Update array_properties.py

In the ArrayPropertyTest class, the lbf actor setup was moved to the test_lb() method, which was the only test using the actor. The actor is now removed after the test. This fixes an issue in test_partial_periodic() when changing the system's periodicity. The error which occured can be seen in event.cpp, function on_cell_structure_change(). This, however, didn't cause the test to fail because the error was not handled by Python. It did cause the test to fail during development of the Stokesian Dynamics functionality, when a handle_errors() was added into globals.pyx.

biermanncarl committed:
Update array_properties.py

KaiSzuttor commented on testsuite/python/array_properties.py:
```diff
@@ -268,12 +266,17 @@ def test_rot_aniso(self):
-        self.locked_operators(self.lbf[0, 0, 0].stress)
-        self.locked_operators(self.lbf[0, 0, 0].stress_neq)
-        self.locked_operators(self.lbf[0, 0, 0].population)
+        self.locked_operators(lbf[0, 0, 0].velocity)
```
why is it prohibited to set a velocity?

KaiSzuttor replied:
and why does this test pass at all? There should only be an error raised if the shape is wrong

https://github.com/espressomd/espresso/blob/b89263db5db6e4084118fc4c54ba0f9068bcb9e5/src/python/espressomd/lb.pyx#L387-L400

biermanncarl replied:
Sorry I do not understand the question. What do you mean by "prohibited to set a velocity"? Where does such an error occur?

biermanncarl replied:
I think I partly understand your question now. But I am not the author of these lines of code, I just adapted them so they would run with the new setup. I cannot answer any questions on why they are there.

KaiSzuttor replied:
I think I misunderstood the test... Probably the test should have some doc-strings what is being tested but you are right that this is not your business.

KaiSzuttor commented on testsuite/python/array_properties.py:
```diff
@@ -268,12 +266,17 @@ def test_rot_aniso(self):
+        self.locked_operators(lbf[0, 0, 0].stress_neq)
+        self.locked_operators(lbf[0, 0, 0].population)
+
+        self.system.actors.clear()
```
in general this should go into the special function ```tearDown```

biermanncarl replied:
The actors.clear() is necessary there because otherwise it would interfere with test_partial_periodic(). The latter test raises a C++ error as long as an LB actor is active.

biermanncarl replied:
I suppose the test_lb() test should go into its own testcase then, because it does not share the same setup with the other tests?

biermanncarl replied:
Discussed this with Jean-Noel. Agreed. Will do that.

KaiSzuttor replied:
Maybe to clarify: the ```tearDown``` method is called automatically after each function that matches ```test*```. See [unittest documentation](https://docs.python.org/3.8/library/unittest.html#unittest.TestCase.tearDown).

biermanncarl replied:
Yep, I first wrongly assumed it would only be called at the very end of all tests.

biermanncarl committed:
Update array_properties.py

Put ```actors.clear()``` into ```tearDown()``` and introduced new ```setUp()``` functions which sets the box size. This is due to the LB actor creation having failed due to too small box size. The LB actor ```lbf``` was now called after ```test_common()``` where the box is resized.

KaiSzuttor commented:
superseded by #3452
